crates/nu-command/src/filters/select.rs

```diff
@@ -346,32 +342,18 @@ impl Iterator for NthIterator {
 
     fn next(&mut self) -> Option<Self::Item> {
         loop {
-            if !self.skip {
-                if let Some(row) = self.rows.first() {
-                    if self.current == *row {
-                        self.rows.remove(0);
-                        self.current += 1;
-                        return self.input.next();
-                    } else {
-                        self.current += 1;
-                        let _ = self.input.next();
-                        continue;
-                    }
-                } else {
-                    return None;
-                }
-            } else if let Some(row) = self.rows.first() {
+            if let Some(row) = self.rows.first() {
                 if self.current == *row {
-                    self.rows.remove(0);
+                    self.rows.pop_first();
```

Comment:
I actually want to ideally rewrite this in terms of the draining Iterator of `BTreeSet` so there is no need to rebalance or drop individually.